### ISSUE TITLE
linkerd dashboard can't connect to linkerd web when node is lost

### ISSUE THREAD

**Issue by falcoriss** [linkerd/linkerd2#3551]
### What is the issue?
linkerd is trying to connect to a pod attached to lost node

### How can it be reproduced?
scale linkerd-web to 2 replicas then crash the node on which 1 pod is. The pod will switch to unknown and k8s will reschedule the pod on another node but the dashboard will keep trying to reach this one.

### Logs, error output, etc

```
$ linkerd dashboard
Error running port-forward: error upgrading connection: error dialing backend: dial tcp 10.0.4.142:10250: connect: no route to host
Check for `linkerd dashboard` running in other terminal sessions, or use the `--port` flag.
```

```
$ kubectl get pods -n linkerd  
NAME                                      READY   STATUS    RESTARTS   AGE
linkerd-controller-5db74fcc6b-d9dkk       3/3     Running   0          42h
linkerd-controller-5db74fcc6b-jxtlp       3/3     Running   0          167m
linkerd-grafana-5cdcb776d6-7dxqg          2/2     Running   0          12m
linkerd-grafana-5cdcb776d6-k6qkq          2/2     Unknown   4          20h
linkerd-grafana-5cdcb776d6-vhbzc          2/2     Running   1          42h
linkerd-identity-68b7d4f7d5-28cmd         2/2     Running   0          42h
linkerd-identity-68b7d4f7d5-2tlvd         2/2     Running   0          12m
linkerd-identity-68b7d4f7d5-6pj7p         2/2     Unknown   4          20h
linkerd-prometheus-6b58985c8c-dhd4w       2/2     Running   0          12m
linkerd-prometheus-6b58985c8c-hp5dd       2/2     Unknown   4          20h
linkerd-prometheus-6b58985c8c-jln9z       2/2     Running   0          167m
linkerd-proxy-injector-5d8c5cb9f4-7qpjj   2/2     Running   0          167m
linkerd-proxy-injector-5d8c5cb9f4-9rnwm   2/2     Running   0          42h
linkerd-sp-validator-7f8947869c-bgnlg     2/2     Running   0          12m
linkerd-sp-validator-7f8947869c-n9dzh     2/2     Running   0          42h
linkerd-sp-validator-7f8947869c-t2kcz     2/2     Unknown   4          20h
linkerd-tap-7ddd67c4b9-58vhb              2/2     Running   0          42h
linkerd-tap-7ddd67c4b9-kcgnr              2/2     Unknown   4          20h
linkerd-tap-7ddd67c4b9-qj7xg              2/2     Running   0          12m
linkerd-web-67d4cb68bc-djnwc              2/2     Unknown   4          20h
linkerd-web-67d4cb68bc-gc8ss              2/2     Running   0          12m
linkerd-web-67d4cb68bc-jpw22              2/2     Running   0          42h
```

#### `linkerd check` output

```text
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ control plane is up-to-date
√ control plane and cli versions match

Status check results are √

```

### Environment

- Kubernetes Version: 1.12.3
- Cluster Environment: kubespray on vmware
- Host OS: ubuntu16.04
- Linkerd version: 2.5 stable

### Possible solution

It seems that the command port-forward to a pod instead of a service. I don't really know how it gets the pod in case there are more than one replica but in my case it was getting the one in unknown state on the lost node. I'm not familiar enough with go to find where it's located on the code, but it should port-forward to the service linkerd-web i believe.

**Comment by grampelberg** [comment]
`linkerd dashboard` does a port forward to the `linkerd-web` service. Internally client-go just uses the service to look up a single pod and forwards all traffic to that. If a pod dies, `dashboard` (and `port-forward`) would need be restarted.

Did it just stop working entirely or did it start working again on restart?

**Comment by ihcsim** [comment]
@falcoriss The tl;dr (https://github.com/kubernetes/kubernetes/issues/15180#issuecomment-457311160) is that even when we issue a port forward call to a service, behind the scene it's still only using a single pod. AFAIK, that behaviour hasn't changed recently.

**Comment by falcoriss** [comment]
> `linkerd dashboard` does a port forward to the `linkerd-web` service. Internally client-go just uses the service to look up a single pod and forwards all traffic to that. If a pod dies, `dashboard` (and `port-forward`) would need be restarted.
> 
> Did it just stop working entirely or did it start working again on restart?

it stopped working entirely, the "linkerd dashboard" command would never be able to connect to the healthy replica, it would keep trying the unknown one on the lost node

**Comment by grampelberg** [comment]
I'm wondering if this is related to #3606 as well.

**Comment by cpretzer** [comment]
@falcoriss in the test that I ran for #3606, I attempted to reproduce this, but did not see the same behavior.

**Comment by cpretzer** [comment]
@falcoriss have you seen this issue on kubernetes 1.17? I'm curious to know if the fix from #3606 also fixed this
